config/v1alpha1/types_backup.go

```diff
@@ -0,0 +1,131 @@
+	// See if the upstream CronJob has CEL validation already defined somewhere.
+	// TODO: Validation to disallow unrealistic schedules (e.g */2 * * * * every 2 mins)
+	// +kubebuilder:validation:Optional
+	// +optional
+	Schedule string `json:"schedule"`
+
+	// The time zone name for the given schedule, see https://en.wikipedia.org/wiki/List_of_tz_database_time_zones.
+	// If not specified, this will default to the time zone of the kube-controller-manager process.
+	// See https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#time-zones
+	// +kubebuilder:validation:Optional
+	// +optional
+	// TODO: CEL validation for timezone format
+	TimeZone string `json:"timeZone,omitempty"`
+
+	// RetentionPolicy defines the retention policy for retaining and deleting existing backups.
+	// +kubebuilder:validation:Optional
+	// +optional
+	RetentionPolicy RetentionPolicy `json: "retentionPolicy"`
+
+	// PVCName specifies the name of the PersistentVolumeClaim which binds a PersistentVolume where the
+	// etcd backup files would be saved
+	// +kubebuilder:validation:Required
+	// +required
+	// TODO: Cannot be empty string
+	PVCName string `json: "pvcName"`
+}
+
+// RetentionType is the enumeration of valid retention policy types
+// +enum
+// +kubebuilder:validation:Enum:="RetentionCount";"RetentionSizeGb"
+type RetentionType string
+
+const (
+	RetentionTypeCount RetentionType = "RetentionCount"
+	RetentionTypeSize  RetentionType = "RetentionSizeGb"
+)
+
+// RetentionPolicy defines the retention policy for retaining and deleting existing backups.
+// This struct is a discriminated union that allows users to select the type of retention policy from the supported types.
+// +union
+type RetentionPolicy struct {
+	// RetentionType sets the type of retention policy. The currently supported and valid values are "retentionCount"
+	// Currently, the only valid policies are retention by count (RetentionCount) and by size (RetentionSizeGb). More policies or types may be added in the future.
+	// +unionDiscriminator
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum:=RetentionCount;RetentionSizeGb
+	RetentionType RetentionType `json:"retentionType"`
+
+	// RetentionCount defines the maximum number of backups to retain.
+	// If the number of successful backups matches retentionCount
+	// the oldest backup will be removed before a new backup is initiated.
+	// The count here is for the total number of backups
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Optional
+	// +optional
+	RetentionCount int `json: "retentionCount,omitempty"`
+
+	// RetentionSizeGb defines the total size in Gb of backups to retain.
+	// If the current total size backups exceeds RetentionSizeGb then
+	// the oldest backup will be removed before a new backup is initiated.
+	// +kubebuilder:validation:Minimum=0.1
+	// +kubebuilder:validation:Optional
+	// +optional
+	RetentionSizeGb float64 `json: "retentionSizeGb,omitempty"`
```

Comment:
> make a struct containing this int please.
> make a struct containing a resource.quantity please.

@deads2k resurfacing your [earlier review](https://github.com/openshift/enhancements/pull/1370#discussion_r1179654101) of the API here again. I can make the change but I'm missing the rationale here so could you please elaborate if it's a convention or requirement to wrap these as structs. Or if it helps with compatibility if we change the API later?

Reply:
Nevermind, found the API convention which points this out.
https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#writing-a-union-in-go